test/Microsoft.Extensions.Primitives.Tests/StringValuesTests.cs

```diff
@@ -263,13 +292,15 @@ public void Contains(StringValues stringValues, string[] expected)
         public void CopyTo(StringValues stringValues, string[] expected)
         {
             ICollection<string> collection = stringValues;
-
             string[] tooSmall = new string[0];
-            Assert.Throws<ArgumentException>(() => collection.CopyTo(tooSmall, 0));
-
             string[] actual = new string[expected.Length];
-            Assert.Throws<ArgumentOutOfRangeException>(() => collection.CopyTo(actual, -1));
-            Assert.Throws<ArgumentException>(() => collection.CopyTo(actual, actual.Length + 1));
+
+            if (collection.Count > 0)
+            {
+                Assert.Throws<ArgumentException>(() => collection.CopyTo(tooSmall, 0));
+                Assert.Throws<ArgumentOutOfRangeException>(() => collection.CopyTo(actual, -1));
+                Assert.Throws<ArgumentException>(() => collection.CopyTo(actual, actual.Length + 1));
```

Comment:
https://github.com/aspnet/Home/wiki/Engineering-guidelines#unit-test-structure

We don't have a concept like "sections" - they should be separate tests.